test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs

```diff
@@ -0,0 +1,133 @@
+
+            var schema = textData.Schema;
+            // First verify that the columns are there. There ought to be at least one column corresponding to the identifiers in the tuple.
+            Assert.True(schema.TryGetColumnIndex("label", out int labelIdx), "Could not find column 'label'");
+            Assert.True(schema.TryGetColumnIndex("text", out int textIdx), "Could not find column 'text'");
+            Assert.True(schema.TryGetColumnIndex("numericFeatures", out int numericFeaturesIdx), "Could not find column 'numericFeatures'");
+            // Next verify they have the expected types.
+            Assert.Equal(BoolType.Instance, schema.GetColumnType(labelIdx));
+            Assert.Equal(TextType.Instance, schema.GetColumnType(textIdx));
+            Assert.Equal(new VectorType(NumberType.R4, 3), schema.GetColumnType(numericFeaturesIdx));
+            // Next actually inspect the data.
+            using (var cursor = textData.GetRowCursor(c => true))
+            {
+                var labelGetter = cursor.GetGetter<DvBool>(labelIdx);
+                var textGetter = cursor.GetGetter<DvText>(textIdx);
+                var numericFeaturesGetter = cursor.GetGetter<VBuffer<float>>(numericFeaturesIdx);
+
+                DvBool labelVal = default;
+                DvText textVal = default;
+                VBuffer<float> numVal = default;
+
+                Assert.True(cursor.MoveNext(), "Could not move even to first row");
+
+                labelGetter(ref labelVal);
+                textGetter(ref textVal);
+                numericFeaturesGetter(ref numVal);
+
+                Assert.Equal(DvBool.False, labelVal);
+                Assert.Equal(new DvText("hello"), textVal);
+                Assert.Equal(3, numVal.Length);
+                Assert.Equal(3.14159f, numVal.GetItemOrDefault(0));
+                Assert.Equal(-0f, numVal.GetItemOrDefault(1));
+                Assert.Equal(2f, numVal.GetItemOrDefault(2));
+
+                Assert.True(cursor.MoveNext(), "Could not move to second row");
+
+                labelGetter(ref labelVal);
+                textGetter(ref textVal);
+                numericFeaturesGetter(ref numVal);
+
+                Assert.Equal(DvBool.True, labelVal);
+                Assert.Equal(new DvText("1"), textVal);
+                Assert.Equal(3, numVal.Length);
+                Assert.Equal(2f, numVal.GetItemOrDefault(0));
+                Assert.Equal(4f, numVal.GetItemOrDefault(1));
+                Assert.Equal(15f, numVal.GetItemOrDefault(2));
+
+                Assert.False(cursor.MoveNext(), "Moved to third row, but there should have been only two");
+            }
+
+            // The next step where we shuffle the names around a little bit is one where we are
+            // testing out the implicit usage of copy columns.
+
+            var est = text.CreateEstimator(r => (text: r.label, label: r.numericFeatures));
+            var newText = text.Append(est);
```

Comment:
>Append [](start = 31, length = 6)

Will there be an `AppendEstimator` ?

Reply:
I think what we decided on is that we could have an `Append` on `Estimator` only, that is like the current `CreateEstimator`. Then out of any of these "schema-bearing" objects we can somehow create a new pipeline, which will be done with a method looking like `Pipline.MakeNew` or something that creates a new empty estimator. Let me know if you do not like it.

---
In reply to: [214167545](https://github.com/dotnet/machinelearning/pull/778#discussion_r214167545) [](ancestors = 214167545)

Reply:
Maybe `Estimator.MakeNew` to avoid promulgation of concepts?

---
In reply to: [214485968](https://github.com/dotnet/machinelearning/pull/778#discussion_r214485968) [](ancestors = 214485968,214167545)